### DISCUSSION TITLE
some problem with pyinstaller

### DISCUSSION THREAD

**Discussion by wxh0402** [Kludex/uvicorn#1852] [Q&A]
fastapi 0.89.1
uvicorn[standard] 0.20.0 (it did not happed on 0.17.6)
python 3.11.1
pyinstaller 5.7.0

This promble only happens when i use pyinstaller package the program with version 0.20.0,and when i reduce program version to 0.17.6,it fix.
![image](https://user-images.githubusercontent.com/50483907/215272465-20ea7f38-c831-48ac-bf4a-a7cfa554c0ee.png)

When i use thwo routers like this,the router [name]with[s] will not get the state 
router config/{name} can get state,but /configs cannot get the state,when i change /configs  to /configlist , it works.

![2L01F 86%W8`LMH6 87M) 3](https://user-images.githubusercontent.com/50483907/215272483-0dafa840-ec35-4c7c-8673-83786d57bb8f.png)

**Comment by Kludex** [comment]
How's this related to uvicorn? :thinking:

**Reply by wxh0402** [reply]
I describes the wrong situation.
It happened when i use uvicorn version >=0.18.0 and package the code with pyinstaller.
I have a  dict store in main process to save some info,and it works correctly when in dev or in 0.17.6 anyway,but when package it with 0.18.0 +  and request multiple apis at the same time,some apis cannot get the store info correctly,like the following image.I am sure it has the value to this key.
![image](https://user-images.githubusercontent.com/50483907/215469960-db380edf-ba43-49dd-9b4e-355cb6f64ac2.png)

I wonder if i make some wrong if I used any wrong usage with the update.
